Components/ProductToShop.php

```diff
@@ -186,12 +186,26 @@ public function commit()
      */
     public function insertOrUpdate(Product $product)
     {
+        $product = $this->eventManager->filter(
+            'Connect_ProductToShop_InsertOrUpdate_Before',
+            $product
+        );
+
         // todo@dn: Set dummy values and make product inactive
         if (empty($product->title) || empty($product->vendor)) {
             return;
         }
 
         $detail = $this->helper->getArticleDetailModelByProduct($product);
+        $detail = $this->eventManager->filter(
+            'Connect_Merchant_Get_Article_Detail_After',
+            $detail,
```

Comment:
@sbofirov: I think this is exactly the purpose of this filter event. ;-)
In some cases you want to merge a non-connect product with an existing article in your shop. Then you can trigger the "merging" with this filter by running your custom "select" in the filter function and return the corresponding detail, but then I would do the `if` in the opponent way: 
`if($detail) {return $detail;} else { /* search article */ }`.
So it's intended that the null-`$detail` is going in the event.

It could be possible to only trigger this event if `$detail` is `null`, but i think this would block the usage of your example. So it would be nice to always trigger this event, but I agree the obligatory `if` on the beginning isn't nice...